pandas/tests/io/json/test_pandas.py

```diff
@@ -192,11 +192,14 @@ def test_roundtrip_str_axes(self, orient, convert_axes, numpy, dtype):
         # to disambiguate whether those keys actually were strings or numeric
         # beforehand and numeric wins out.
         # TODO: Split should be able to support this
-        if convert_axes and (orient in ("split", "index", "columns")):
+        if convert_axes and (orient in ("index", "columns")):
             expected.columns = expected.columns.astype(np.int64)
             expected.index = expected.index.astype(np.int64)
         elif orient == "records" and convert_axes:
             expected.columns = expected.columns.astype(np.int64)
+        elif convert_axes and orient == "split":
+            expected.columns = expected.columns.astype(np.int64)
+            expected.index = expected.index.astype(str)
```

Comment:
Do we need this conversion? I think the problem is that we are converting when we don't need to